src/connection-manager/auto-dialler.ts

```diff
@@ -37,6 +39,8 @@ export class AutoDialler implements Startable {
   private readonly options: Required<AutoDiallerInit>
   private running: boolean
   private autoDialTimeout?: ReturnType<retimer>
+  private readonly abortController?: AbortAbort
+  connectionMap: Map<string, Promise<Connection>>
```

Comment:
You can use [PeerMap](https://libp2p.github.io/js-libp2p-peer-collections/classes/PeerMap.html) from `@libp2p/peer-collections` - no need to stringify the peer ids then.